kakao/KakaoLoginAndroid/app/src/main/java/com/google/firebase/auth/kakao/MainActivity.java

```diff
@@ -0,0 +1,217 @@
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.auth.kakao.databinding.ActivityMainBinding;
+import com.kakao.auth.ISessionCallback;
+import com.kakao.auth.Session;
+import com.kakao.usermgmt.LoginButton;
+import com.kakao.usermgmt.UserManagement;
+import com.kakao.usermgmt.callback.LogoutResponseCallback;
+import com.kakao.util.exception.KakaoException;
+import com.kakao.util.helper.log.Logger;
+
+import android.os.Handler;
+import android.widget.Toast;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class MainActivity extends AppCompatActivity {
+
+    LinearLayout loggedInView;
+    LoginButton loginButton;
+    Button logoutButton;
+    NetworkImageView imageView;
+
+    ActivityMainBinding binding;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        initUI();
+        Session.getCurrentSession().addCallback(new KakaoSessionCallback());
+    }
+
+    private void initUI() {
+        setContentView(R.layout.activity_main);
+        Toolbar toolBar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolBar);
+
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
+        loggedInView = (LinearLayout) findViewById(R.id.logged_in_view);
+        loginButton = (LoginButton) findViewById(R.id.login_button);
+        logoutButton = (Button) findViewById(R.id.logout_button);
+        imageView = (NetworkImageView) findViewById(R.id.profile_image_view);
+
+        logoutButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                UserManagement.requestLogout(new LogoutResponseCallback() {
+                    @Override
+                    public void onCompleteLogout() {
+                        FirebaseAuth.getInstance().signOut();
+
+                        Handler handler = new Handler(Looper.getMainLooper());
+                        handler.post(new Runnable() {
+                            @Override
+                            public void run() {
+                                updateUI();
+                            }
+                        });
+                    }
+                });
+            }
+        });
+        updateUI();
+    }
+
+    /**
+     * Update UI based on Firebase's current user. Show Login Button if not logged in.
+     */
+    private void updateUI() {
+        FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
+        if (currentUser != null) {
+            binding.setCurrentUser(currentUser);
+            if (currentUser.getPhotoUrl() != null) {
+                imageView.setImageUrl(currentUser.getPhotoUrl().toString(), new ImageLoader(Volley.newRequestQueue(getApplicationContext()), new ImageLoader.ImageCache() {
```

Comment:
Since this is just a sample it is probably easier to import Glide and do:

```
Glide.with(this)
  .load(currentUser.getPhotoUrl)
  .into(imageView)
```

Much shorter, which means developers can pay more attention to the important parts of this sample (the authentication)!